### PR TITLE
[FEATURE] More unified and improved error handling in Adapters.

### PR DESCRIPTION
**Pull request by jpfeuffer**
- Use of QTempDirectory with auto remove functionality if debug level <2
- Always parse std err and std out if exit code != 0
- More unified logging

One could even go further and abstract external process creation/logging as well as improve the temp file handling in TOPPBase (e.g. to cleanup in the destructor).

### CODE REVIEW

**Comment by hroest** [comment]
looks good but tons of duplicated code. Maybe we can put some of that into `OpenMS/APPLICATIONS/TOPPBase.h` or even a new class `OpenMS/APPLICATIONS/AdapterBase.h` ?

**Comment by jpfeuffer** [comment]
Ok I did it. Although I also liked the AdapterBase idea more, I think it is not worth it yet for the one function that I had in mind.

**Comment by timosachsenberg** [comment]
+1

File: src/openms/source/APPLICATIONS/TOPPBase.cpp [hunk 1]
```diff
@@ -1609,6 +1627,36 @@ namespace OpenMS
...
+    if (success == false || qp.exitStatus() != 0 || qp.exitCode() != 0)
+    {
+      writeLog_("FATAL: External invocation of " + executable.toStdString() + " failed. Standard output and error were:");
+      const QString stdout(qp.readAllStandardOutput());
+      const QString stderr(qp.readAllStandardError());
+      writeLog_(stdout);
+      writeLog_(stderr);
+      writeLog_(String(qp.exitCode()));
+
+      return EXTERNAL_PROGRAM_ERROR;
```
Comment (hroest) [inline src/openms/source/APPLICATIONS/TOPPBase.cpp]: do you catch this error?

Comment (jpfeuffer) [reply src/openms/source/APPLICATIONS/TOPPBase.cpp]: puh.. no.. I just want the program to exit. I see that it does not work anymore.

Comment (hroest) [reply src/openms/source/APPLICATIONS/TOPPBase.cpp]: yes, but now this returns from the function, not from `main` any more

**Pull request merged**